src/CMakeLists.txt

```diff
@@ -5,11 +5,30 @@
 #                                                                              #
 ################################################################################
 
-# @author Nicolo Genesio <nicolo.genesio@iit.it> 
+# @author Nicolo Genesio <nicolo.genesio@iit.it> ,Luca Tricerri <luca.tricerri@iit.it>
+
+include(FetchContent)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+option(COMPILE_WITHUNITTEST "Compile also unittest" OFF)
+
+# The unit tests uses io_service::run_for that is available from boost 1.66
+if(COMPILE_WITHUNITTEST AND Boost_VERSION_STRING GREATER_EQUAL 1.66.0)
+  FetchContent_Declare(
+    unittest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG        origin/master
```

Comment:
To ensure reproducible results given a commit of the `blocktest` repo, I would suggest to use a tag of the googletest repository, instead of using a branch.

Reply:
Ok done @traversaro